src/aaz_dev/cli/controller/portal_cli_generator.py

```diff
@@ -0,0 +1,359 @@
+        return (None, None, None, None)
+
+    def fill_cmd_examples(self, cmd_info, cmd_cfg, leaf, target_version):
+        var_option_list = []
+        var_required = set()
+        subId_required = False
+        for arg_group in cmd_cfg.arg_groups:
+            for arg in arg_group.args:
+                assert arg.options is not None
+                if len(arg.options) == 0: continue
+                is_required = arg.required
+                var_name = arg.var.replace("$Path", "$path")
+                if arg.required and var_name.find("subscriptionId") == -1:
+                    # for user input cmd example, ignore subscriptionId
+                    var_required.add(var_name)
+                if arg.required and var_name == "$path.subscriptionId":
+                    subId_required = True
+                if cmd_info['path'].find("{resourceGroupName}") != -1 and var_name == "$path.resourceGroupName":
+                    var_required.add(var_name)
+                    is_required = True
+                var_option_list.append((var_name, arg.options,
+                                        is_required,
+                                        var_name.find('$path') != -1))
+
+        if target_version.examples is None or len(target_version.examples) == 0:
+            self.generate_default_example(cmd_info, leaf, var_option_list)
+            return
+        find_example = False
+        for example in target_version.examples:
+            if find_example: break
+            assert example.name is not None
+            assert example.commands is not None
+            for command_str in example.commands:
+                if find_example: break
+                param_verify = True
+                para_list = []
+                para_groups = re.finditer(self.PARA_REG_PATTERN, command_str)
+                if not para_groups:
+                    continue
+                cmd_para = set()
+                # 1. check if all parameters in example command str is in path, if not ignore
+                for para_match in para_groups:
+                    para_name = para_match.group(1)
+                    var, raw_options, is_required, is_in_path = self.find_para_var(para_name, var_option_list)
+                    # var not found, drop this command_str
+                    if not var or not is_in_path:
+                        param_verify = False
+                        break
+                    pick_options = list(filter(lambda opt: len(opt) > 1, raw_options))
+                    para_list.append((var, pick_options))
+                    cmd_para.add(var)
+
+                # 2. check if all required parameters is in example command str
+                # if not, drop this example command str
+                if not var_required.issubset(cmd_para):
+                    param_verify = False
+                if not param_verify: continue
+                parameters = []
+                subId_added = False
```

Comment:
please use snack case
```suggestion
                sub_id_added = False
```